.github/workflows/publish-gh-pages.yml

```
@@ -0,0 +1,54 @@
+name: Publish Github Pages
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        #python version is required in setup the python environment
+        required: true
+        type: string
+
+jobs:
+  publish-docs:
+    name: "Publish Github Pages"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      # Checkout the code
+      - name: Checkout Source
+        uses: actions/checkout@v2
+        with:
+          # require all of history to see all tagged versions' docs
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ inputs.python-version }}
+
+      - name: Install Dependencies
+        run: |
+          python3 -m pip install poetry
+          poetry install
+
+      - name: Checkout gh-pages
+        uses: actions/checkout@v2
+        with:
+          ref: gh-pages
+          path: docs/_build
+
+      - name: Build Docs
+        run: |
+          cd docs
+          poetry run sphinx-multiversion source/ _build/
+          touch _build/.nojekyll
```

Nice. However, this would bonk if `_build/.nojekyll` already exists. I recommend replacing with

```suggestion
          rm _build/.nojekyll && touch _build/.nojekyll
```

---

`touch` will not fail if the file already exists.  It just updates the file's last updated date.